src/app/[locale]/projects/detail/[id]/components/ViewClearingRequestModal.tsx

```diff
@@ -0,0 +1,214 @@
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+
+// SPDX-License-Identifier: EPL-2.0
+// License-Filename: LICENSE
+
+'use client'
+
+import { signOut, useSession } from "next-auth/react"
+import { useTranslations } from "next-intl"
+import { Dispatch, SetStateAction, useCallback, useEffect, useState } from "react"
+import { Alert, Button, Form, Modal } from "react-bootstrap"
+import { BsCheck2Square } from "react-icons/bs"
+import { ClearingRequestDetails, HttpStatus } from "@/object-types"
+import { ApiUtils } from "@/utils/index"
+import { notFound } from "next/navigation"
+import Link from "next/link"
+import styles from '@/app/[locale]/requests/requestDetail.module.css'
+
+interface Props {
+    show?: boolean
+    setShow?: Dispatch<SetStateAction<boolean>>
+    projectName?: string
+    clearingRequestId?: string
+}
+
+export default function ViewClearingRequestModal({ show,
+                                                   setShow,
+                                                   projectName,
+                                                   clearingRequestId }: Props) {
+    const t = useTranslations('default')
+    const [message, setMessage] = useState('')
+    const { data: session, status } = useSession()
+    const [variant, setVariant] = useState('success')
+    const [showMessage, setShowMessage] = useState(false)
+    const [clearingRequestData, setClearingRequestData] = useState<ClearingRequestDetails>({
+        id: '',
+        requestedClearingDate: '',
+        projectId: '',
+        projectName: '',
+        requestingUser: '',
+        projectBU: '',
+        requestingUserComment: '',
+        clearingTeam: '',
+        agreedClearingDate: '',
+        priority: '',
+        clearingType: '',
+        reOpenOn: null,
+        createdOn: '',
+        comments: [{}],
+        _embedded: {
+            "sw360:project": {
+                name: '',
+                version: ''
+            }
+        }
+    })
+
+    const displayMessage = (variant: string, message: string) => {
+        setVariant(variant)
+        setMessage(message)
+        setShowMessage(true)
+    }
+
+    const fetchData = useCallback(
+        async (url: string) => {
+            const response = await ApiUtils.GET(url, session.user.access_token)
+            if (response.status == HttpStatus.OK) {
+                const data = await response.json() as ClearingRequestDetails
+                return data
+            } else if (response.status == HttpStatus.FORBIDDEN) {
+                displayMessage('warning',  t('Failed to fetch clearing request from database'))
+                return signOut()
+            } else if (response.status == HttpStatus.UNAUTHORIZED) {
+                return signOut()
+            } else {
+                notFound()
+            }
+        },[session]
+    )
+
+    useEffect(() => {
+        void fetchData(`clearingrequest/${clearingRequestId}`).then(
```

Comment:
should check session is not null or undefined before fetching data

Reply:
Addressed the issue.